From: Bernd Zeimetz <bernd@bzed.de>
Date: Sat, 17 Aug 2019 01:27:08 +0200
Subject: [PATCH] Fix merging and using of python *flags.

As follow-up for d0e0864c2802860ff561fe0b39939b63d38b8c70,
also fix the handling of python build flags.
---
 SConstruct | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/SConstruct b/SConstruct
index bf682a03e087..d71dfab48db3 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1169,7 +1169,7 @@ else:
 
 PYTHON_LIBDIR_CALL = 'sysconfig.get_python_lib()'
 
-PYTHON_CONFIG_NAMES = ['CC', 'CXX', 'OPT', 'BASECFLAGS',
+PYTHON_CONFIG_NAMES = ['CC', 'CXX', 'OPT', 'CFLAGS',
                        'CCSHARED', 'LDSHARED', 'SO', 'INCLUDEPY', 'LDFLAGS']
 PYTHON_CONFIG_QUOTED = ["'%s'" % s for s in PYTHON_CONFIG_NAMES]
 PYTHON_CONFIG_CALL = ('sysconfig.get_config_vars(%s)'
@@ -1753,14 +1753,17 @@ else:
     ldshared = ldshared.replace('-fPIE', '')
     ldshared = ldshared.replace('-pie', '')
     python_env.Replace(SHLINKFLAGS=[],
-                       LDFLAGS=python_config['LDFLAGS'],
+                       LDFLAGS=[],
                        LINK=ldshared,
                        SHLIBPREFIX="",
                        SHLIBSUFFIX=python_config['SO'],
                        CPPPATH=[python_config['INCLUDEPY']],
-                       CPPFLAGS=python_config['OPT'],
-                       CFLAGS=python_config['BASECFLAGS'],
-                       CXXFLAGS=python_config['BASECFLAGS'])
+                       CPPFLAGS=[],
+                       CFLAGS=[],
+                       CXXFLAGS=[])
+
+    for flag in ['CFLAGS','LDFLAGS','OPT']:
+        python_env.MergeFlags(Split(python_config[flag]))
 
     python_objects = {}
     python_compiled_libs = {}
